include/orb/Orbit.h

```diff
@@ -0,0 +1,391 @@
+GNC_TRACKED_CONSTANT(constexpr geograv::Coeff<PANGRAVORDER>, PANGRAVITYMODEL, static_cast<geograv::Coeff<PANGRAVORDER>>(GGM05S));
+
+GNC_TRACKED_CONSTANT(const double, MAXORBITRADIUS, 6378.0E3L+1000.0E3);
+
+GNC_TRACKED_CONSTANT(const double, MINORBITRADIUS, 6378.0E3L);
+
+
+/**
+ * Class to handle most of the logic related to storing, serializing, propagating, and checking validity of an orbit.
+ */
+class Orbit {
+  public:
+    /** Position of the sat (m).*/
+    lin::Vector3d _recef= lin::nans<lin::Vector3d>();
+
+    /** Velocity of the sat (m/s).*/
+    lin::Vector3d _vecef= lin::nans<lin::Vector3d>();
+
+    /** Time since gps epoch (ns).*/
+    uint64_t _ns_gps_time{0};
+
+    /** Validity of the orbit.
+     * A valid orbit has finite and real position and velocity, is in low
+     * earth orbit, and has a reasonable time stamp (within 20 years of pan epoch).
+     * The validity check should not reject 
+     * gps readings due to reasonable noise of:
+     * TODO add max expected gps error see issue #372
+     *
+     * Low earth orbit is a Orbit that stays between MINORBITRADIUS and MAXORBITRADIUS.
+     */
+    bool _valid{false};
+
+    /** Returns time since gps epoch (ns).
+     * The Orbit must be not propagating.
+     * 
+     * grav calls: 0 */
+    uint64_t nsgpstime() const{
+        return _ns_gps_time;
+    }
+
+    /** Return position of the sat (m).
+     * The Orbit must be not propagating.
+     * 
+     * grav calls: 0 */
+    lin::Vector3d recef() const{
+        return _recef;
+    }
+
+    /** Return velocity of the sat (m/s).
+     * The Orbit must be not propagating.
+     * 
+     * grav calls: 0 */
+    lin::Vector3d vecef() const{
+        return _vecef;
+    }
+
+    /** Return true if the Orbit is valid.
+     * A valid orbit has finite and real position and velocity, is in low
+     * earth orbit, and has a reasonable time stamp (within 20 years of pan epoch).
+     * The validity check should not reject 
+     * gps readings due to reasonable noise of:
+     * TODO add max expected gps error see issue #372
+     *
+     * Low earth orbit is a Orbit that stays between MINORBITRADIUS and MAXORBITRADIUS.
+     * 
+     * grav calls: 0 */
+    bool valid() const{
+        return _valid;
+    }
+    
+    /** 
+     * Helper function calculate if the Orbit is valid, see valid().
+     * If the orbit is invalid set the orbit to the default values.
+     * 
+     * grav calls: 0 */
+    void _check_validity(){
+        double r2= lin::fro(_recef);
+        //TODO add checks for time and velocity see issue #372
+        //note if position is NAN, these checks will be false.
+        if (r2<MAXORBITRADIUS*MAXORBITRADIUS && r2>MINORBITRADIUS*MINORBITRADIUS){
+            _valid= true;
+        } else {
+            _valid= false;
+            _recef= lin::nans<lin::Vector3d>();
+            _vecef= lin::nans<lin::Vector3d>();
+            _ns_gps_time= 0;
+        }
+    }
```

Comment:
This totally works and is. Just as an aside for future reference, if you wanted to ensure all elements of a `lin` object are finite you can use `lin::all(lin::isfinite(v))`.